AmazonPublisherServicesAdapter/src/main/java/com/chartboost/mediation/amazonpublisherservicesadapter/AmazonPublisherServicesAdapter.kt

```diff
@@ -823,15 +732,17 @@ class AmazonPublisherServicesAdapter : PartnerAdapter {
         return (partnerAd.ad)?.let { ad ->
             (ad as? DTBAdInterstitial)?.let {
                 suspendCancellableCoroutine { continuation ->
-                    fun resumeOnce(result: Result<PartnerAd>) {
-                        if (continuation.isActive) {
-                            continuation.resume(result)
-                        }
-                    }
+                    val continuationWeakRef = WeakReference(continuation)
 
                     onShowSuccess = {
                         PartnerLogController.log(SHOW_SUCCEEDED)
-                        resumeOnce(Result.success(partnerAd))
+                        continuationWeakRef.get()?.let {
```

Comment:
oh then again there are only weak references here, so it should be fine

Reply:
yep, nothing here so should be good